### PR TITLE
Recognize Indent and Dedent as a Token

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Here is the Output Comparison:
Python vs LPython
Comparison

### CODE REVIEW

File: tests/reference/tokens-indent1-290e858.stdout [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+(TOKEN ">") 93:93
+(TOKEN "integer" 0) 95:95
+(TOKEN ":") 96:96
+(NEWLINE) 97:97
+(TOKEN "indent") 98:105
+(KEYWORD "print") 106:110
+(TOKEN "(") 111:111
+(TOKEN ")") 112:112
+(NEWLINE) 113:113
+(TOKEN "dedent") 114:114
```
Comment (certik) [inline tests/reference/tokens-indent1-290e858.stdout:41]: I think this should be 2 `dedent` tokens, but it only emits 1.

File: tests/reference/tokens-indent1-290e858.stdout [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+(TOKEN ">") 63:63
+(TOKEN "integer" 0) 65:65
+(TOKEN ":") 66:66
+(NEWLINE) 67:67
+(TOKEN "indent") 68:75
+(KEYWORD "print") 76:80
+(TOKEN "(") 81:81
+(TOKEN ")") 82:82
+(NEWLINE) 83:83
+(TOKEN "dedent") 84:87
```
Comment (certik) [inline tests/reference/tokens-indent1-290e858.stdout:29]: This correctly emits 1 `dedent` token, so all good.

File: tests/reference/tokens-indent1-290e858.stdout (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply tests/reference/tokens-indent1-290e858.stdout]: Yes sir, it should be 2 dedent. Now I fixed it.

**Comment by Thirumalai-Shaktivel** [comment]
This is ready. Any other changes to be made, @certik??

File: src/lpython/parser/tokenizer.h [hunk 3]
```diff
@@ -23,6 +23,10 @@ class Tokenizer
     bool enddo_newline_process = false;
     int enddo_state = 0;
     int enddo_insert_count = 0;
+    bool indent = false;
```
Comment (certik) [inline src/lpython/parser/tokenizer.h:26]:
```suggestion
    bool indent = false; // Next line is expected to be indented
```

**Pull request merged**